### PR TITLE
WIP: This change allows [snippets] to be placed on the same line as m…

### PR DESCRIPTION
**Pull request by uglow**
…arkdown (not sure about HTML).

However, the [render snippet args] snippet uses the render.ejs template, which normally wraps the content
in a <div class="row">. It would be nice to be able to create documentation-HTML snippets that
do not need to be isolated or wrapped in anything else other than the documentation classes available, or
the snippet's own *.styl classes.

### CODE REVIEW

**Comment by rodleviton** [comment]
I have had a bit of a think about this and I am starting to think that isolating the css for the snippets and app specific css might be overkill. The idea of a pattern library theme should be to complement the patterns, components etc. The theme is a lightweight shell that is also scoped for with unique classnames. Root level html items like fonts, h1...h6, paragraph tags etc would be the only concern for style clashes but I think ost pattern libraries would want these to be consistent in the theme as well.

If an isolated region is really a concern a special [isolate-] tag could be introduced that uses a shadow DOM to isolate css - not a 100% sure how you would extract or determine the css that should be loaded into here and not the root as well?

@uglow interested in your thoughts here.

**Comment by uglow** [comment]
I think I agree with you. The isolation of CSS should only be necessary for rendering components, **if the components are not using CSS Modules themselves**. Historically, this would be the case. Going forward, more components are going to have an interface like this:

```
<my-component icon-class="bootstrap-class1" button-class="btn btn-primary"> ...
```

So the components will provide properties to allow developers to override default styles for certain sub-components of the component (e.g. an icon or a button in the above example). In this way, a component can have internal styles (using CSS Modules) and also allow certain styles to be injected from the "global" stylesheet.

In summary, I think making isolation configurable on a per-component basis is necessary, as you've proposed.

**Comment by uglow** [comment]
This is a good article explaining how ShadowDOM custom elements can still be themed: http://www.html5rocks.com/en/tutorials/webcomponents/shadowdom-201/
